resource-rating-plugin.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Plugin Name: Resource Rating
+ * Description: A plugin to enable resource rating on posts and pages within WordPress
+ * Version: 0
+ * Author: Annabel Baynes
+ */
+
+//----------------------------------------------------------------------------------------------------------------------
+
+/**
+ * Description: Returns all of the resource ratings present on a post (identified by the post id sent through as a parameter).
+ *
+ * @param $post_id
+ *
+ * @return mixed
+ */
+function get_the_resource_ratings ($post_id) {
+	return get_post_meta($post_id, "resource_ratings");
+}
+
+/**
+ * Description: Adds a new rating to a post if it is a valid integer (post id and rating passed as parameters).
+ *
+ * @param $post_id
+ * @param $rating_value
+ */
+function submit_new_resource_rating ($post_id, $rating_value) {
+	if (filter_var($rating_value, FILTER_VALIDATE_INT)) {
+		add_post_meta( $post_id, "resource_ratings",$rating_value, false );
+	} else {
+		echo "Input is not an integer!";
+	}
+}
+
+/**
+ * Description: Deletes all resource ratings present on a post (identified by the post id sent through as a parameter).
+ *
+ * @param $post_id
+ */
+function delete_all_resource_ratings ($post_id) {
```

Comment:
I'm not sure on what basis would the data be deleted. I guess there is still some more work to be done with this.